ke/asm.py

```diff
@@ -300,6 +300,123 @@ class NegativeLookbehind(ParensSyntax):
 Lookbehind.INVERTED = NegativeLookbehind
 
 
+class InlineFlag(namedtuple("InlineFlag", ["flag_character", "name", "sub"]), Asm):
+    A = "a"
+    L = "L"
+    U = "u"
+    I = "i"
+    M = "m"
+    S = "s"
+    regex_to_kleenexp = {
+        A: "ascii_only",
+        L: "locale_dependent",
+        U: "unicode",
+        I: "ignore_case",
+        M: "multiline",
+        S: "any_matches_all",
+    }
+    UNSET = "unset"
+    UNSETTABLE = set([I, M, S])
+    INCOMPATIBLE = set([A, L, U])
+    # Unicode flag CANNOT be used in bytes-like,
+    # locale_dependent can ONLY be used in bytes-like.
+    # This value is being set before compilation start.
+    PATTERN_IS_BYTES_LIKE = False
+
+    def to_regex(self, flavor, capture_names, wrap=False):
+        # Here we break the tree structure, using this specialized
+        # function to create a flagging expression for the rest of
+        # the subtree. The only flag in a flagging sequence that
+        # reaches the to_regex() method is the first one in the
+        # sequence (there may be other sequences). This is necessary
+        # because the parenthesis wrapping and regex legality are
+        # dependent on the whole flagging expression.
```

Comment:
If we drop all the validations, there is always the option to simply compile ```[ignore_case multiline 'string']``` to ```(?i:(?m:string))``` instead of ```(?im:string)```